src/core/load-fragment.c

```diff
@@ -4749,7 +4749,18 @@ int config_parse_exec_directories(
                                 continue;
                 }
 
-                r = exec_directory_add(ed, sresolved, dresolved);
+                ExecDirectoryFlags exec_directory_flags = 0;
+                if (!isempty(flags)) {
+                        if (exec_directory_flags_from_string(flags) != EXEC_DIRECTORY_READ_ONLY) {
```

Comment:
hmm, why not write:

```c
exec_directory_flags = exec_directory_flags_from_string(flags);
if (exec_directory_flags < 0 || (exec_directory_flags & ~_EXEC_DIRECTORY_PUBLIC) != 0) {
        log_syntax(…
```

i.e. let's write this generically so that new flags are added later easily, without requiring rewriting this. it's also more obvious